### ISSUE TITLE
Compile the generated client in Javascript

### ISSUE THREAD

**Issue by vlidholt** [serverpod/serverpod#1874]
Explore if it is possible to use Dart JS compiler to generate a client for Javascript. An advantage here is that the client can then be used on the web or in React.

**Comment by easazade** [comment]
Have you considered generating a typescript client? Not sure it would be possible to use dart web interop to do that.

**Comment by Isakdl** [comment]
Yes! We have discussed this, I think it would make sense if we create a bespoke JS client to also add the type definitions.

The compiling route as discussed in this issue turned out to be pretty difficult, and would in the end mean that we more or less would have to define the entire interface manually anyway.

**Comment by ChristopherLinnett** [comment]
Thought this could be useful for some side projects, so threw this together today with help from Claude. 

Have only done a little bit of testing so far, but works with my current projects.

https://github.com/ChristopherLinnett/serverpod_typescript_bridge

**Comment by RunnerDu67** [comment]
Could it be possible to create a second folder like my_project_generator and let users create their onw generator for their needs. 
It could generate files like yaml for endpoints and models and we could simply use this files to generate custom clients for python, nodejs, react, vue, angular, etc.

And with a future version of the CLI include the generators with a config in the generator.yaml that let us choose whitch ones execute and maybe even use serverpod without flutter or dart client
